src/IO/GEHDF5Wrapper.cxx

```diff
@@ -228,9 +395,13 @@ Succeeded GEHDF5Wrapper::initialise_exam_info()
     unsigned int num_time_slices = 0;
     H5::DataSet timeframe_dataspace = file.openDataSet("/HeaderData/SinglesHeader/numValidSamples");
     timeframe_dataspace.read(&num_time_slices, H5::PredType::NATIVE_UINT32);
+
+    if(is_list_file())
+        m_num_singles_samples=num_time_slices;
```

Comment:
`initialise_exam_info` (this function) is called always at `open()`, we can not enforce the file type.

Reply:
ok, then this shouldn't be in this function. Its location a remnant of the erroneous use of `TimeFrameDefinitions`

Reply:
I can read it in the listmode initializer. Do you want me to get rid of the rest of the time_frame stuff too?

Reply:
As currently the `num_singles` isn't stored in `exam_info`, it is confusing to me if we read it here.

The time frame definitions constructed here are not correct. They are appropriate for the singles only (and only in listmode I presume). I haven't checked where they are used, but at least for the moment, I'd move it there, until we are sure how it is with non-list files.

Instead, time-frame-defs should be read from `HeaderData/AcqStats`, i.e. .`frameStartTime` and `frameDuration`. So, construct a `vector` with a single `std::pair` with `frameStartTime` and `frameStartTime+frameDuration/1000` I believe.


Also, set 
````
exam_info.start_time_in_secs_since_1970=double(header.sharc_rdf_acq_stats_data.scanStartTime
```

Reply:
Haven't really followed this last comment sorry. 

Sure, `num_singles_samples` goes into listmode initializer. 

I can move the code for the time frames somewhere else, but its stored in `exam_info_sptr`.
Do you mean I should move them somewhere else, but also cosntruct them differently, with data from somewhere else (AcqStats)?

Also that last line, I don't know how to implement it. What is `header`?

Reply:
the current time-frame code is incorrect. It gives 1s time frames because that is what the singles sampling rate is. @pw351 did that to be able to pass it to the `SinglesRates*` object I guess (I haven't  checked if it's actually used).

the proposed code sets correct time frame info. This new code should remain here.

Final line, I copied some stuff and forgot to edit. I meant
```
exam_info.start_time_in_secs_since_1970=double(scanStartTime);
```
with the latter also read from the `AcqStats` dataset.

If thiiss itoo confusing, let me know and I'll do this myself.

Reply:
Ah yes, it was used like that, but I changed it now, and it reads from `num_singles_samples` instead.

I'll give this a go.

Reply:
So, some issue, maybe. 

the sino and listmode data do return the same value  for `frameStartTime`, but `norm3d` and `geo3d` do not (a different one each). Is this expected/OK?

Reply:
On top of that, it seems that frameStartTime is also since 1970, should I subtract that time? Not sure if the floats are storing these large numbers well.

Reply:
> the sino and listmode data do return the same value for frameStartTime, but norm3d and geo3d do not (a different one each). Is this expected/OK?

expected. The norm comes from a specific acquisition on a different date.

>  seems that frameStartTime is also since 1970, should I subtract that time? 

ok, please subtract `scanStartTime` then.